components/feature/sendtab/src/main/java/mozilla/components/feature/sendtab/SendTabUseCases.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.components.feature.sendtab
+
+import androidx.annotation.VisibleForTesting
+import mozilla.components.concept.sync.Device
+import mozilla.components.concept.sync.DeviceCapability
+import mozilla.components.concept.sync.DeviceConstellation
+import mozilla.components.concept.sync.DeviceEventOutgoing.SendTab
+import mozilla.components.concept.sync.TabData
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import mozilla.components.support.ktx.kotlin.crossProduct
+
+/**
+ * Contains use cases for sending tabs to devices related to the firefox-accounts.
+ */
+class SendTabUseCases(accountManager: FxaAccountManager) {
+
+    class SendToDeviceUseCase internal constructor(
+        private val accountManager: FxaAccountManager
+    ) {
+        /**
+         * Sends the tab to provided deviceId if possible.
+         *
+         * @param deviceId The deviceId to send the tab.
+         * @param tab The tab to send.
+         * @return a deferred boolean of the result.
+         */
+        suspend operator fun invoke(deviceId: String, tab: TabData) = sendAsync(deviceId, tab)
+
+        /**
+         * Sends the tabs to provided deviceId if possible.
+         *
+         * @param deviceId The deviceId to send the tab.
+         * @param tabs The list of tabs to send.
+         * @return a deferred boolean if the combined result was successful or not.
+         */
+        suspend operator fun invoke(deviceId: String, tabs: List<TabData>): Boolean {
+            return tabs.map { tab ->
+                sendAsync(deviceId, tab)
+            }.fold(true) { acc, result ->
+                acc and result
+            }
+        }
+
+        private suspend fun sendAsync(deviceId: String, tab: TabData): Boolean {
```

Comment:
Here and elsewhere:
The convention for methods with the `Async` suffix is that they return a Deferred value. How about just `send`? It's already a suspend function.

Reply:
Fixed! I've left the `Async` suffix on only the use case bindings since those are the only ones that need it now.